### ISSUE TITLE
Not able to run multiple Qase Test Cases in parallel

### ISSUE THREAD

**Issue by mdiep-asp** [qase-tms/qase-java#140]
Currently, if we try to run multiple Qase Test Cases in parallel (using maven surefire plugin + testng) by splitting all test cases into 3 test suites, we will see this error:

```
[main] INFO com.app.appium.provider.DeviceProvider - The test device is: cloud_ios
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Activating app
[main] INFO com.app.appium.core.BaseTest - Test: testABC , Result:failed
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Terminating app
[main] INFO com.app.appium.provider.DeviceProvider - The test device is: cloud_ios
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Activating app
[main] ERROR io.qase.api.CasesStorage - Previous case is still in progress.
[main] INFO com.app.appium.core.BaseTest - Test: testDEF , Result:failed
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Terminating app
[main] INFO com.app.appium.provider.DeviceProvider - The test device is: cloud_ios
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Activating app
[main] ERROR io.qase.api.CasesStorage - Previous case is still in progress.
[main] INFO com.app.appium.core.BaseTest - Test: testGHI , Result:failed
[main] INFO com.app.appium.utilities.AppiumDriverUtils - Terminating app
```

Upon looking into Qase api, we saw this:

<img width="727" alt="Screenshot 2024-07-02 at 2 30 13 PM" src="https://github.com/qase-tms/qase-java/assets/81988166/6c921b61-b7d0-4c72-b49d-5ad2615eaccb">

Seems like each test case run is associated with a thread. Even though we run the other test suites with a different maven instance, somehow the existing running test case thread from one suite was being called by others and caused this issue. 

Wonder Qase actually supports running in parallel? If not, will Qase soon support it? Please help. Thanks

**Comment by mdiep-asp** [comment]
@savkk we ran our mobile automation tests on CircleCI with these configs:

- We broke the entire test suite into a 3 smaller test suites
- Each smaller test suite was running on its own distinct workflow of CircleCI
- We used maven surefire plugin to execute these smaller test suites

We believe that even though there are 3 running distinct workflows. They are still on the same machine, and probably share the same threads pool.

**Comment by savkk** [comment]
@mdiep-asp, have you set up parallel run through testng? Could you give an example of testng configuration?

**Comment by mdiep-asp** [comment]
@savkk we didn't set up parallel run thru testng. We basically have 3 open terminals, each run with command like this `mvn clean test -DtestngXmlFile=$PARAM_TESTNG_XML -DSuiteName=$PARAM_SUITE_NAME` in parallel.

**Comment by savkk** [comment]
> @savkk we didn't set up parallel run thru testng. We basically have 3 open terminals, each run with command like this `mvn clean test -DtestngXmlFile=$PARAM_TESTNG_XML -DSuiteName=$PARAM_SUITE_NAME` in parallel.

It looks really strange. If you run tests in different processes, they should not affect each other. They have their own memory.
If you run a single process, does this error not appear?

**Comment by gibiw** [comment]
@mdiep-asp Hi! Try to update the TestNg reporter to the latest version 4.0.0.

**Comment by mdiep-asp** [comment]
Hi @gibiw, looks like qase-testng 4.0.0 not available on Maven yet?

**Comment by gibiw** [comment]
@mdiep-asp You need to use `qase-testng-reporter` package

**Comment by mdiep-asp** [comment]
thanks @gibiw it looks like that it does fix the above issue. However, I may need to open another issue. For some reasons, with the same configuration, the new SDK seems to automatically update the test cases' title and suite. Is there anyway to avoid this?
<img width="541" alt="Screenshot 2024-10-31 at 7 04 27 PM" src="https://github.com/user-attachments/assets/3b976a46-9cd8-454d-8f16-199aa6b23bc7">

**Comment by gibiw** [comment]
@mdiep-asp This is not really a problem. You can manage updates in the project settings.
![image](https://github.com/user-attachments/assets/c594daf2-f321-449f-8e0b-edc77d266699)

**Comment by mdiep-asp** [comment]
@gibiw thank you so much, really appreciate your guidance
